xbmc/addons/interfaces/GUI/controls/Label.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ *      Copyright (C) 2005-2017 Team Kodi
+ *      http://kodi.tv
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with KODI; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "Label.h"
+#include "addons/kodi-addon-dev-kit/include/kodi/gui/controls/Label.h"
+
+#include "addons/AddonDll.h"
+#include "guilib/GUILabelControl.h"
+#include "guilib/GUIWindowManager.h"
+#include "utils/log.h"
+
+extern "C"
+{
+namespace ADDON
+{
+
+void Interface_GUIControlLabel::Init(AddonGlobalInterface* addonInterface)
+{
+  addonInterface->toKodi->kodi_gui->control_label = static_cast<AddonToKodiFuncTable_kodi_gui_control_label*>(malloc(sizeof(AddonToKodiFuncTable_kodi_gui_control_label)));
+
+  addonInterface->toKodi->kodi_gui->control_label->set_visible = set_visible;
+  addonInterface->toKodi->kodi_gui->control_label->set_label = set_label;
+  addonInterface->toKodi->kodi_gui->control_label->get_label = get_label;
+}
+
+void Interface_GUIControlLabel::DeInit(AddonGlobalInterface* addonInterface)
+{
+  free(addonInterface->toKodi->kodi_gui->control_label);
+}
+
+void Interface_GUIControlLabel::set_visible(void* kodiBase, void* handle, bool visible)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUILabelControl* control = static_cast<CGUILabelControl*>(handle);
+  if (!addon || !control)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIControlLabel::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, control, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  control->SetVisible(visible);
+}
+
+void Interface_GUIControlLabel::set_label(void* kodiBase, void* handle, const char *label)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUILabelControl* control = static_cast<CGUILabelControl*>(handle);
+  if (!addon || !control || !label)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIControlLabel::%s - invalid handler data (kodiBase='%p', handle='%p', label='%p') on addon '%s'",
+                          __FUNCTION__, addon, control, label, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  // create message
+  CGUIMessage msg(GUI_MSG_LABEL_SET, control->GetParentID(), control->GetID());
+  msg.SetLabel(label);
+
+  // send message
```

Comment:
Becomes removed. Always this c&p things :-D 
https://github.com/xbmc/xbmc/blob/master/xbmc/interfaces/legacy/Control.cpp#L139-L147